### PR TITLE
✨ Source Microsoft SharePoint: Add parsing shared files

### PR DESCRIPTION
**Pull request by tolik0**
## What

- Add fetching of shared files and folders. Resolves: https://github.com/airbytehq/airbyte-internal-issues/issues/2504
- Fix errors with file paths.
- Migrate to poetry.

## How
Iterating over shared objects is implemented using the requests library, as office365 doesn't support getting items using drive ID and object ID.
Add a new parameter, search_scope, to select where to search for the files.

Fix file paths by adding the drive URL to the file path to differentiate between files from different drives.

### CODE REVIEW

File: airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/stream_reader.py [hunk 1]
```diff
@@ -92,14 +106,65 @@ def config(self, value: SourceMicrosoftSharePointSpec):
...
+                else:  # Object is a folder, retrieve children
+                    child_url = f"{base_url}/items/{child['id']}/children"  # Use item endpoint for nested objects
+                    yield from get_files(child_url, new_path)
+            yield from []
+
+        # Initial request to item endpoint
+        item_url = f"{base_url}/items/{object_id}"
+        item_response = requests.get(item_url, headers=headers)
+        if item_response.status_code != 200:
+            raise RuntimeError(f"Error retrieving shared object: {item_response.status_code}")
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/stream_reader.py:148]: We need specified more user friendly error message with more details.

**Review by lazebnyi — APPROVED** [review]
lgtm 

minor comment for error message

File: airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/stream_reader.py (hunk 1 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/stream_reader.py]: Fixed

**Review by lazebnyi — APPROVED** [review]
lgtm

**Pull request merged**